sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -276,25 +280,30 @@ export abstract class V4PositionManager {
 
     // If migrating, we need to settle and sweep both currencies individually
     if (isMint(options) && options.migrate) {
-      // payer is v4 positiion manager
+      if (options.useNative) {
+        planner.addUnwrap(BigNumber.from('0x8000000000000000000000000000000000000000000000000000000000000000'))
```

Comment:
this should be defined as a constant

Reply:
also its not always the case you will want to unwrap the full amount no? maybe it complicates things but by unwrapping the full amount refunds from a weth pool will be in eth

Reply:
could change to `planner.addUnwrap(BigNumber.from(0))` and keep the weth sweep at the end

Reply:
that way it gets just the contract's debt instead of the whole balance, and refunds could be kept in weth

Reply:
the trading api calculates the liquidity to add based off of the 2 amounts received from v3, so im not sure when the contract balance wouldnt equal the correct amount tho. but could do this for good measure

Reply:
yeah i think it makes sense to do it with OPEN_DELTA instead

Reply:
do we not allow partial liquidity deposits?

Reply:
even if not there could 100% be a case where because of price differences the amount of weth in the v3 position > the amount of eth required in the v4 position

Reply:
when migrating u need to migrate 100%